pkg/api/query/grpc.go

```diff
@@ -158,6 +140,47 @@ func (g *GRPCAPI) Query(request *querypb.QueryRequest, server querypb.Query_Quer
 	return nil
 }
 
+func (g *GRPCAPI) getQueryForEngine(ctx context.Context, engineParam querypb.EngineType, request *querypb.QueryRequest, queryable storage.Queryable, maxResolution int64) (promql.Query, error) {
```

Comment:
I think `engineParam` is already in the request, so we don't need it as an argument.